tools/nginz_disco/nginz_disco.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+set -e
+exec 2>&1
+
+# Assumes /etc/wire/nginz/conf and /etc/wire/nginz/upstreams directories exist
+# Assumes a file upstreams.txt containing space-separated dns names.
+
+upstream_list="/etc/wire/nginz/conf/upstreams.txt"
+old="/etc/wire/nginz/upstreams/upstreams.conf"
+new="${old}.new"
+
+function valid_ip() {
```

Comment:
How about having a `valid_ipv4` and `valid_ipv6`... ?

Reply:
Done.